pkg/mutation/conversion_test.go

```diff
@@ -287,13 +297,15 @@ func TestParseShouldFail(t *testing.T) {
 			Namespace: "namespace",
 		},
 		Spec: mutationsv1alpha1.AssignMetadataSpec{
-			Match:      mutationsv1alpha1.Match{},
-			Location:   "spec...foo",
-			Parameters: mutationsv1alpha1.MetadataParameters{},
+			Match:    mutationsv1alpha1.Match{},
+			Location: "spec...foo",
+			Parameters: mutationsv1alpha1.MetadataParameters{
+				Assign: runtime.RawExtension{Raw: []byte(`{"value": "foobar"}`)},
+			},
 		},
 	}
 	_, err = mutation.MutatorForAssignMetadata(assignMeta)
-	if err == nil || !strings.Contains(err.Error(), "parse") {
+	if err == nil || !strings.Contains(err.Error(), "invalid location format") {
```

Comment:
nit: maybe I missed this somewhere do we also want a test for assign value field to make sure it's string for metadata vs interface for assign?

Reply:
this code should do that:

https://github.com/open-policy-agent/gatekeeper/blob/0370f150244c96dc48d60430d018d6ea82a14d3c/pkg/mutation/assignmeta_mutator.go#L174-L180

Which is moved to `MutatorForAssignMetadata` as part of this PR

Reply:
I meant do we have a unit test for this

Reply:
actually I see this is tested in  pkg/webhook/mutator_validation_test.go  so we can ignore this.

Reply:
yeah, probably best to test where it should be enforced. Hard to know the correct balance between coverage, safety and avoiding repetition. Thanks for the review!